sample/src/main/java/com/vimeo/sample/network/NetworkRequest.java

```diff
@@ -116,16 +120,17 @@ void cancelRequest() {
                     builder.append(line);
                 }
 
+                Stag.Factory factory = new Stag.Factory();
                 Gson gson = new GsonBuilder().registerTypeAdapter(Date.class, new DateParser())
-                        .registerTypeAdapterFactory(new Stag.Factory())
+                        .registerTypeAdapterFactory(factory)
                         .create();
 
                 JsonObject jsonObject = new JsonParser().parse(builder.toString()).getAsJsonObject();
 
+                TypeAdapter<VideoList> videoListTypeAdapter = factory.getVideoList$TypeAdapter(gson);
                 long time = System.currentTimeMillis();
 
-                videos.addAll(gson.fromJson(jsonObject, VideoList.class).data);
-
+                videos.addAll(videoListTypeAdapter.fromJsonTree(jsonObject).data);
```

Comment:
This will not be able to handle generic classes. 

This is however a good idea for a future release. We have created a helper class which does something similar : It does not have a switch case though. We can possibly generate a helper class which does the fromJson thing. 

We should do this in a separate class because this class has become too big. For our case where there are 350+model classes the loading of this class is taking 15ms